Frontiersmen Trench Complex \\ Budgetcuts disabling

Luftkommando opened the pull request:
## About The Pull Request

Adds a Frontiersman-centric ruin to the rock-planet lineup. Frontiersmen Trench Complex has a primarily combat-based loot pool but also should take advantage of the PR'd [working ruin subshuttle](https://github.com/shiptest-ss13/Shiptest/pull/5207) with a Brawler-class landed on the landing zone. 

Also removes Budgetcuts.

<details><summary>Areas</summary>
<img width="2560" height="1760" alt="StrongDMM-2025-08-20 09 16 43" src="https://github.com/user-attachments/assets/b232d760-ffc2-441b-8bcf-8edac8de2081" />
</details>


<details><summary>Fullbright</summary>
<img width="2560" height="1760" alt="StrongDMM-2025-08-20 09 16 45" src="https://github.com/user-attachments/assets/f5f6dbb8-eb98-4327-aafc-7b466742830d" />
</details>

<details><summary>Natural Lighting</summary><img width="540" height="540" alt="1080p" src="https://github.com/user-attachments/assets/c73abb19-199e-4f55-9e55-088512be55b0" /></details>

## Why It's Good For The Game

Diversity in ruins is good, and dead Inteq means independents can get their hands on Inteq specific weapons for possible in-round tension if the player-led Inteq want to press their claim on 'their' equipment or not and Inteq can get additional weaponry of their faction at a cost of raw time instead of money. Removing budgetcuts probably hurts the diversity note for the Trench Complex but it's an old ruin and full of loot that probably shouldn't be around anyway.

## Changelog

:cl:
add: Added a new ruin, an entrenched Frontiersman group, to the Rock Planet ruin pool.
del: Removed Budgetcuts from the ruin pool.
/:cl:

Luftkommando committed:
Create somme.dmm \\ Disable budgetcuts.dmm

Creates a new rock planet ruin called Frontiersmen Trench Complex and disables the budgetcuts ruin.

Luftkommando committed:
Check Fixing

Capitalizes all the rooms that I just didn't capitalize for some reason. \\ Removes the subshuttle dock for the moment because I'm not 100% on what specifically went wrong since it looks identical in the .dmm's text as ship shuttle docks and should fit without landing on any walls. \\  Removes the double shutters and dirt hiding in the wall.

Erikafox commented on code/datums/ruins/rockplanet.dm:
```diff
@@ -5,12 +5,13 @@
 	id = "rockplanet_budgetcuts"
 	suffix = "rockplanet_budgetcuts.dmm"
 	ruin_tags = list(RUIN_TAG_HARD_COMBAT, RUIN_TAG_MEDIUM_LOOT, RUIN_TAG_LIVEABLE)
+*/
```
Just remove it and the map

Erikafox commented on code/game/area/areas/ruins/rockplanet.dm:
```diff
@@ -156,3 +156,40 @@
+
+/area/ruin/rockplanet/somme/trench
+	name = "Trenchline"
+	icon_state = "hallb"
```
needs a valid icon_state

Erikafox reviewed (changes requested):
This is a pretty decent start for a first-time map
<hr>
Overall, you need to do a lot more variation in the tiles and decals. I would say look at other ruins using Frontiersman (beach_depot and rockplanet_distillery come to mind) to get an idea of what their interiors and exteriors look like. Concrete everything is an idea in the same vein of the bacon everything breakfast, it sounds really good until you try it.

I would consider adding a turret here and there to the trench line. It'll present a static threat as opposed to the frontiersmen, who will run when engaged.
<hr>
<img width="872" height="606" alt="image" src="https://github.com/user-attachments/assets/e4d7a387-7436-4e84-ad07-a73fe0985371" />
The LZ should be outside and exposed to the atmosphere. I would extend the map and make a trench-surrounded landing pad
<img width="1081" height="973" alt="image" src="https://github.com/user-attachments/assets/a3f704cc-916c-461c-be7c-34fa81d4f10c" />
Very vague mock-up of what it could be
<hr>
<img width="1214" height="577" alt="image" src="https://github.com/user-attachments/assets/67afd49e-840a-4ea6-9d18-cff89e560074" />
I would try and work more hard angles into this
<hr>
I am also of the opinion that you can add more spindles and outcroppings to the trenches. Little elevated forward posts, a little spotter nest. Nests for the turrets, so forth.

<hr>
<img width="588" height="736" alt="image" src="https://github.com/user-attachments/assets/ba7a2550-16f2-460d-8568-7a3edb27f4ef" />
This is free loot hidden inside a hill. Please remove it.

<hr>
<img width="447" height="735" alt="image" src="https://github.com/user-attachments/assets/04cf9b44-8dd6-4dae-8f42-62d6173b389b" />
Once again I would look at how other maps design their spaces to get an idea of how this'd look. I would also recommend being comfortable with making hard angles on the interior. The nature of byond means that they're usually "better".
<img width="491" height="546" alt="image" src="https://github.com/user-attachments/assets/9557d4a9-5620-471d-a65c-6514ea744c61" />
See how the top of this room feels when it's not holding that weird angle for the closet?
<hr>
<img width="768" height="893" alt="image" src="https://github.com/user-attachments/assets/e3e55f6b-7f16-4cd3-8604-a24e09bc0c2a" />
Don't be afraid to add decals for the sort of room an area is supposed to be.
<hr>
<img width="916" height="761" alt="image" src="https://github.com/user-attachments/assets/0c8f4d31-a903-442a-8d92-627fae39bf0c" />
This is one of those weirder spaces. I would change where the entrance is and make the machinery in a corner 
<hr>
<img width="1052" height="676" alt="image" src="https://github.com/user-attachments/assets/804a9d55-2696-4576-9c07-a86eedfe6473" />
This section is very awkward. Once again harder angles and demarkating out what something should be with decals will help.
<hr>
<img width="1074" height="753" alt="image" src="https://github.com/user-attachments/assets/92941168-3bc3-45fb-9d07-ebe1f5986048" />
Once you move the landing pad, I think this area can be a very nice warehouse/staging section.
<hr>
<img width="186" height="219" alt="image" src="https://github.com/user-attachments/assets/425b9056-ba8b-4430-852e-c12f5342e123" />
I would make a guy with the wasp who'll always drop it. Maybe a new subtype of the officer. It'd be a fun boss.


<hr>
<hr>
Important question. How do they sleep if the entire base is filled with badmos

Luftkommando committed:
First attempt at addressing criticisms

Removes Budgetcuts.dmm, its area, and its entry in rockplanet.dm.

Rearranges the Frontiersman Trench Complex and extends it in the X direction to accommodate a true outdoors landing pad and the surrounding trench. Adds few extrusions with turrets and small mud-trench lines to get to them.

The interior should now be atmos-safe, with faux-airlocks similar to tesla lab. In my head they just slept with their oxy cans on because they're buffoons like that but it's probably better to do proper atmos. The LZ, now a warehouse, has holofields because of it.

Makes a new Frontiersman Boss subtype, the Frontiersman Captain, who has a wasp. +1 on the burst but extra spread compared to the trooper version. Also adds internals and neutered versions of the Frontiersman Commander, the rifle wielding boss, because I thought it would be nice for consistency.

To keep Ross in line with Commander Tiyak as a 'custom' boss I created the code\modules\ruins\rockplanet_ruin_code\somme.dm and stuck him in there with the 100% drop chance. Randomname gen on death is fine since Ross is probably a pseudonym anyway.

Luftkommando committed:
Compile Issues

Dirty walls are unacceptable.

Also forgot to load the somme.dm

Luftkommando committed:
Fixes Areas and encircles the LZ

Forgot to make a change to the Trenchline area to accompany a change in the shape of the outbound trenches and circled the entire LZ platform so even the most foolish Frontiersman can recognize that the 'stand clear' goes all the way. Also changes the wood type in a few sections to provide better room-to-room distinction.

Luftkommando committed:
I've got a jar of dirt

Dirt's no longer all up in the walls. Also a light I accidentally placed in a rock wall at some point.

Luftkommando committed:
Shelf oopsie

Forgot to put the crate shelves back to 3. Had them on 4 tall because it looks more like you need a forklift to get up there when they're up against a wall to the North. With them moved to have an open North the four talls look ugly.

Luftkommando committed:
Inadeqaute

Inadequate

Luftkommando committed:
Add blood + trash \\ Separates the airlocks and trenches.

I'm the trash man. I throw garbage everywhere.

Erikafox reviewed (changes requested):
<img width="1001" height="928" alt="image" src="https://github.com/user-attachments/assets/cd26f4e8-3711-4106-9caa-27a8a618b0af" />
Few things with this building.

First off, now that you've gotten it working on a holodoor, there should probably be an alternative interior access method. A side door somewhere along the building would be a nice excuse to make the trenchline move a bit

Wood is a very light tile type for an industrial area. I would change the used tiles to something more solid. Imagine what would happen if a forklift rolled onto the wood

You can also make the bodybags more of a bespoke segment of the interior. I'd just push the trench around a little to get the space. Make it a like. Strip + Sort kinda spot
<hr>
<img width="810" height="579" alt="image" src="https://github.com/user-attachments/assets/a3fba98b-2501-4ae0-a329-4475b74c1743" />
Do something evil. Every frontiersman that's posted directly on a sandbag should have their retreat distance var set to "2"

<hr>
I would also slightly reduce the threat of the turrets. I think the mulcher is in a fine spot, but you may want to add pounders in place of some of the spitters.

<hr>

![StrongDMM_PHpTm83Zsv](https://github.com/user-attachments/assets/5b1242b4-f0f9-4f3c-9103-861b63037337)

This floor safe needs to be telegraphed.

Luftkommando committed:
Second Round of Criticism

Added a non-holofield entry to the warehouse that's locked to begin with by a shutter over the door. The override button for it is in Ross' quarters same as the cargo shutters used to be.

The Inteq corpses are now a bit more organized, insofar as not just being lined up against the wall, and are in the process of being stripped for equipment. The sneaksuit is because I think it's cute and undifferentiated except by sprite but I can put it to a normal Inteq uniform if it's TOO cute.

I considered making a 'trench' subtype in somme.dm for the retreat variable change since other people might want to use it as well, but decided that would clog up the prefab list too much and just varedited them.

A couple of the turrets (E, NE, SW) were changed from spitter to pounder. E and NE to give an avenue of 'less resistance' for return visitors and SW so it can act as a 'hey this ruin exists and has danger' marker for crews.

Floors in the warehouse section were changed to dark tile and the section with the fuel tank was changed to be an exo recharger with a broken power-loader since the warehouse would ICly be getting and sending regular raid and supply shipments.

Let's see if I put dirt into any walls again.

Luftkommando committed:
Dirt

I am a fool, a buffoon, ctrl+2 fails my ailing eyes. Gets rid of the wall dirt and also puts a corner where it belongs.

Luftkommando committed:
Uncuts Corners

I blame undiagnosed ADD/ADHD.

Erikafox reviewed (dismissed):
the frontiersmen intended to be on the walls of the trench should have their "wander" var set to 0

Erikafox commented:
<img width="227" height="172" alt="image" src="https://github.com/user-attachments/assets/012dd3e7-63f4-475d-947f-bdc48b665d40" />
This gas tank is not connected

<img width="582" height="442" alt="image" src="https://github.com/user-attachments/assets/d175916d-987e-4bbf-96a4-6d56df05496e" />
move the vent up a tile

<img width="362" height="343" alt="image" src="https://github.com/user-attachments/assets/f106278c-3861-4cac-8367-e08be542aff7" />
I would re-arrange this so the blast doors start open, the desk has some glass (the computer should be moved out of the way), and make it so the Boss can fire through the glass at people trying to dart by

Luftkommando committed:
Map changes + Frontiersman Sentries

Changes sections of the Trench Complex according to feedback. Adding several entrances to the trenchline and removal of shanks. Also puts the air pump+manifold into the correct order. The Armoury now starts open, and Ross has windows.

All the Frontiersmen who should be roaming the trenches should now have their structure_smash set to 0, so instead of hitting and getting stuck on the platforms they seem, emphasis on the word, to simply pass through them now.

A commissioner is replaced by a Mongrel. There will be no ammo for it since the Frontiersmen don't have a 10mm gun that they would reasonably stock.

Creates a 'new' Frontiersman type, the Frontiersman Sentry. It's just a Frontiersman Sharpshot but they have a longer engagement range to reflect a 'proper sniper' and won't move at all. Essentially a fleshy turret if you think about it a little. This was mostly done because for some reason vision_range would reset from 14 to the generic type's 9 whenever the snipers picked and killed/lost a target. If I knew better how this worked I'd probably change it at the ground level but setting it to their own bespoke type worked in my testing. (Technically they only begin to engage at 13 tiles, not 14. But if you step into engagement and back out to 14 tiles they'll still shoot. I assume it's a discrepancy in how the distances are calculated).

Luftkommando committed:
Some Fixing

I think I forgot to replace the sentries somehow. Fixed their positioning and removed one that I think would make an overlapping sniper fire lane in the SW, which I feel wouldn't be particularly fun.

Luftkommando committed:
Misc fixes + fax fixes

Replaces the frontiersman faxes on both icwbase.dmm and somme.dmm with ruin faxes. Oopsie.

Also finally actually fixes the air tank + some crate issues from positioning.

Erikafox committed:
Merge remote-tracking branch 'upstream/master' into pr/5227

Zevotech commented:
#5198 is going to be merged soon so once you update your branch run the updatepaths script to swap out the deprecated dirt tiles on your map.

<img width="626" height="267" alt="image" src="https://github.com/user-attachments/assets/7ef083df-83ae-41e5-b7f8-9e367d9b2186" />

to do this, click and drag the update_dirt_paths.txt file onto the Update Paths batch file and let it complete, then open your map and check for errors. Do not open your map before doing this or it will delete all of the dirt tiles.

Luftkommando committed:
Space and Time

Fixed the space tile on the binocular rack near the Sniper Redoubt and moved the cover sniper a bit to both 'aim' down the central trench as well as get aimed at through the central trench.

Luftkommando committed:
Merge branch 'master' of https://github.com/shiptest-ss13/Shiptest into shiptest-ss13-master

Luftkommando committed:
Merge branch 'shiptest-ss13-master' into frontiersman_bunker

Luftkommando committed:
5198 dirt paths

I... Think I have done as commanded. Everything looks fine.

Luftkommando committed:
Fixes ICW Base I guess?

Apparently replacing the fax on ICW base deleted the TComm room's walls? I don't know how this happened? It wasn't even all no diagonal walls it was JUST the no diagonal walls that weren't otherwise otherwise occupied by things like turrets and what have you. I'm befuddled and confused. I don't know how adding the fax caused this but it must have because it's in the commit as well and I didn't think too much of it at the time.

thearbiber reviewed (dismissed):
<img width="156" height="175" alt="image" src="https://github.com/user-attachments/assets/cbdf1a93-be3e-47f6-902f-96736e2e293c" />

these doors here dont have their ID's all set correctly. Fix that and it should be good to send.

Luftkommando committed:
Fix Blast Door ID

I had this fixed but forgot to commit it lmao. Fixes the accidental airlock id on the cargo shutters.
